File: everything/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
   long cardnumber = get_long ("card number: ");
   int digitcounter = 1;
   int luhm;
   int digit13;
   int digit14;
   int digit15;
   int digit16;
   for(int i = 0; i < cardnumber; i *= 10)
   {
      long digit = cardnumber % 10;
      cardnumber = cardnumber / 10;
      if (digitcounter == 13)
      {
         digit13 = digit;
      }
      if (digitcounter == 14)
      {
         digit14 = digit;
      }
      if (digitcounter == 15)
      {
         digit15 = digit;
      }
      if (digitcounter == 16)
      {
         digit16 = digit;
      }
      if (digitcounter % 2 == 0)
      {
         digit = digit * 2;
         luhm = luhm + (digit % 10);
         if ((digit % 100) >= 10)
         {
            luhm = luhm + ((digit % 100)/ 10);
         }
      }
      else
      {
         luhm = luhm + digit;
      }
      digitcounter++;
   }
   if (luhm % 10 > 0)
   {
      printf("INVALID\n");
   }
   if (luhm % 10 == 0)
   {
      if (digitcounter - 1 == 15)
      {
         if ((digit14 == (4 | 7)) && (digit15 == 3))
         {
            printf("AMEX\n");
         }
         else
         {
            printf("INVALID\n");
         }
      }
      else if (digitcounter - 1 == 16)
      {
         if ((digit15 > 0 && digit15 < 6) && (digit16 == 5))
         {
            printf("MASTERCARD\n");
         }
         else if (digit16 == 4)
         {
            printf("VISA\n");
         }
         else
         {
            printf("INVALID\n");
         }
      }
      else if (digitcounter - 1 == 13)
      {
         if (digit13 == 4)
         {
            printf("VISA\n");
         }
         else
         {
            printf("INVALID\n");
         }
      }
      else
      {
         printf("INVALID\n");
      }
   }
}

File: everything/sentimental-credit/credit.py
# TODO
from cs50 import get_string

number = get_string("number: ")

luhm = 0
for i in range(len(number) - 2, -1, -2):
    digit = int(number[i]) * 2
    luhm = luhm + (digit % 10)
    luhm = luhm + int(digit / 10)
for i in range(len(number) - 1, -1, -2):
    luhm = luhm + int(number[i])

if luhm % 10 == 0:
    if len(number) == 15:
        if ((int(number[0]) == 3) and (int(number[1]) == 4) or (int(number[1]) == 7)):
            print("AMEX")
        else:
            print("INVALID")

    elif len(number) == 16:
        if ((int(number[0]) == 5) and ((int(number[1]) > 0) and (int(number[1]) < 6))):
            print("MASTERCARD")
        elif int(number[0]) == 4:
            print("VISA")
        else:
            print("INVALID")

    elif len(number) == 13:
        if int(number[0]) == 4:
            print("VISA")
        else:
            print("INVALID")
            
    else:
        print("INVALID")
else:
    print("INVALID")

